controllers/spacebindingcleanup/spacebinding_cleanup_controller.go

```diff
@@ -89,23 +97,31 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return ctrl.Result{}, errs.Wrapf(err, "unable to get the bound Space")
 	}
 
-	murName := types.NamespacedName{Namespace: spaceBinding.Namespace, Name: spaceBinding.Spec.MasterUserRecord}
-	mur := &toolchainv1alpha1.MasterUserRecord{}
-	if err := r.Client.Get(ctx, murName, mur); err != nil {
-		if errors.IsNotFound(err) {
-			logger.Info("the MUR was not found", "MasterUserRecord", spaceBinding.Spec.MasterUserRecord)
-			requeueAfter, err := r.deleteSpaceBinding(ctx, spaceBinding)
-			return ctrl.Result{
-				RequeueAfter: requeueAfter,
-			}, err
+	// check MUR existence.
+	// this check is skipped if the PublicViewer feature is enabled
+	// and the SpaceBinding is related to the PublicViewer.
+	if !r.isPublicViewerEnabled(ctx) || spaceBinding.Spec.MasterUserRecord != toolchainv1alpha1.KubesawAuthenticatedUsername {
+		murName := types.NamespacedName{Namespace: spaceBinding.Namespace, Name: spaceBinding.Spec.MasterUserRecord}
```

Comment:
minor - I find it easier to read and it reflects more the comment :
```
if r.isPublicViewerEnabled(ctx) && spaceBinding.Spec.MasterUserRecord == toolchainv1alpha1.KubesawAuthenticatedUsername {
    return ctrl.Result{}, nil
}
```

Reply:
done in 3a53065078dbc9c811bf76da9379cf4d1145788c